sources/packages/backend/apps/api/src/services/application-offering-change-request/application-offering-change-request.service.ts

```diff
@@ -400,4 +428,56 @@ export class ApplicationOfferingChangeRequestService {
       return applicationOfferingChangeRequest;
     });
   }
+
+  /** Validates student authorization for the provided application offering change request.
+   * @param applicationOfferingChangeRequestId application offering change request id.
+   * @param studentId student id for the authorization.
+   * @returns true if the application offering change request belongs to the student, otherwise false.
+   */
+  async validateStudentForOfferingChangeRequest(
+    applicationOfferingChangeRequestId: number,
+    studentId: number,
+  ): Promise<boolean> {
+    const { application } = await this.applicationOfferingChangeRequestRepo
+      .createQueryBuilder("applicationOfferingChangeRequest")
+      .select(["applicationOfferingChangeRequest.id", "application.id"])
+      .innerJoin("applicationOfferingChangeRequest.application", "application")
+      .innerJoin("application.student", "student")
+      .where(
+        "applicationOfferingChangeRequest.id = :applicationOfferingChangeRequestId",
+        {
+          applicationOfferingChangeRequestId,
+        },
+      )
+      .andWhere("student.id = :studentId", { studentId })
+      .getOne();
+
+    if (application) return true;
+    return false;
+  }
+
+  /**
+   * Update the application offering change request status for the given applicationId.
+   * @param applicationOfferingChangeRequestId application offering change request id for which to update the status.
+   * @param applicationOfferingChangeRequestStatus application offering change request status to be updated.
+   * @param options method options:
+   * - `studentConsent` student consent to approve the application offering change request.
+   */
+  async updateApplicationOfferingChangeRequestStatus(
```

Comment:
https://github.com/bcgov/SIMS/pull/2196/files#r1298923868

for the reference of doing validation here